Service name prefix (#11)
Zooz/prometheus-api-metrics: README.md

- durationBuckets - Buckets for response time in seconds. `default: [0.001, 0.005, 0.015, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5]`
- requestSizeBuckets - Buckets for request size in bytes. `default: [5, 10, 25, 50, 100, 250, 500, 1000, 2500, 5000, 10000]`
- responseSizeBuckets - Buckets for response size in bytes. `default: [5, 10, 25, 50, 100, 250, 500, 1000, 2500, 5000, 10000]`
- useUniqueHistogramName - Add to metrics names the project name as a prefix (from package.json)

> - metricsPrefix - A custom matrics names prefix, the package will add underscode between your prefix to the metric name.

Reviewer: ```suggestion
- metricsPrefix - A custom metrics names prefix, the package will add underscore between your prefix to the metric name.
```

> - excludeRoutes - Array of routes to exclude. Routes sould be in your framework syntax

Reviewer: ```suggestion
- excludeRoutes - Array of routes to exclude. Routes should be in your framework syntax.
```

curl http[s]://<host>:[port]/metrics.json
```

**Note:**

> 1. If you pass to the middleware the `metricsPath` option the path will be the one the you choosed.

Reviewer: ```suggestion
1. If you pass to the middleware the `metricsPath` option the path will be the one that you chose.
```

> 2. If you are using express framework and no route was found for the request (e.g: 404 status code), the request will not be collected. that's because we'll risk in a memory leak since the route is not a pattern but a hardcoded string.

Reviewer: ```suggestion
2. If you are using express framework and no route was found for the request (e.g: 404 status code), the request will not be collected. that's because we'll risk memory leak since the route is not a pattern but a hardcoded string.
```

### Note
This will work only if you use the default Prometheus registry - do not use `new Prometheus.Registry()`

## Request.js HTTP request duration collector

> This feature enbale you to easily process the result of Request.js timings feature.

Reviewer: ```suggestion
This feature enbales you to easily process the result of Request.js timings feature.
```

### Usage
In order to use this feature you must use `{ time: true }` as part of your request configuration and then pass to the collector the response or error you got.

For Example:

#### request
```js
request({ url: 'http://www.google.com', time: true }, (err, response) => {
    Collector.collect(err || response);
});
```

#### request-promise-native
```js
return requestPromise({ method: 'POST', url: 'http://www.mocky.io/v2/5bd9984b2f00006d0006d1fd', route: 'v2/:id', time: true, resolveWithFullResponse: true }).then((response) => {
    Collector.collect(response);
}).catch((error) => {
    Collector.collect(error);
});
```

### Configuration
- durationBuckets - the histogram buckets for request duration.

> - countClientErrors - Boolean that indicates if to collect client errors as Counter, this counter will have target and error code labels.

Reviewer: ```suggestion
- countClientErrors - Boolean that indicates whether to collect client errors as Counter, this counter will have target and error code labels.
```

- useUniqueHistogramName - Add to metrics names the project name as a prefix (from package.json)

> - prefix - A custom matrics names prefix, the package will add underscode between your prefix to the metric name.

Reviewer: ```suggestion
- prefix - A custom metrics names prefix, the package will add underscore between your prefix to the metric name.
```

**Notes:** 
1. In order to use the timing feature in request-promise/request-promise-native you must also use `resolveWithFullResponse: true`

> 2. Override - you can override the `route` and `target` attribute instead of taking them from the request object. In order to do that you should set a `metrics` object on your request with those attribute:

Reviewer: ```suggestion
2. Override - you can override the `route` and `target` attributes instead of taking them from the request object. In order to do that you should set a `metrics` object on your request with those attribute:
```

``` js
request({ method: 'POST', url: 'http://www.mocky.io/v2/5bd9984b2f00006d0006d1fd', metrics: { target: 'www.google.com', route: 'v2/:id' }, time: true }, (err, response) => {...};
});
```

> 

Reviewer: I think you are missing examples of initialising the packages explicitly with `init()` & `collect()` vs `new MetricsCollector(...)`, and explain when you should can use the latter. or maybe i missed it!
